### ISSUE TITLE
use post-install hook to run bootstrap jobs

### ISSUE THREAD

**Issue by clkao** [InfuseAI/primehub#6]
bootstrap jobs including

- create a realm
- create a client for jupytherhub and store the secret
- create a default user with profiles

should be idempotent

## implementations

- [x] post-install hook to bootstrap
- [x] idempotent (if {realm, client, user} has already created, skip to setup it)

## known issues

- user should remove unmanaged `primehub-secret` secret
- helm upgrade might break keycloak in-memory storage deployment (no admin created)

**Comment by qrtt1** [comment]
except `idempotent`, bootstrap is implemented.

```
$ curl -s https://infuseai.github.io/helm-chart/index.yaml
apiVersion: v1
entries:
  primehub:
  - apiVersion: v1
    created: 2019-04-23T05:45:45.386317352Z
    description: PrimeHub Community Edition
    digest: b1ed33fa6c47f81534602bd4f3390f1b444500ca837de2e40056fc90da6bd41f
    home: https://infuseai.io/primehub
    keywords:
    - jupyter
    maintainers:
    - email: hi@infuseai.io
      name: InfuseAI Inc.
    name: primehub
    sources:
    - https://gitlab.com/infuseai/charts/primehub
    tillerVersion: '>=2.11.0'
    urls:
    - https://InfuseAI.github.io/helm-chart/primehub-0.0.1-2214b31.tgz
    version: 0.0.1-2214b31
generated: 2019-04-23T05:45:45.382931782Z
```

## usage

preq

1. a cluster
2. helm and tiller installed
3. set a IP alias [see our settings guide](https://github.com/InfuseAI/primehub/blob/master/CONTRIBUTING.md#setting-up-development-environment) and ingress

### primehub

install a development pagkage with `--devel` flag

```
helm repo add primehub https://infuseai.github.io/helm-chart
helm upgrade --install --namespace primehub primehub primehub/primehub --devel \
  --values https://raw.githubusercontent.com/InfuseAI/primehub/master/config-kind.yaml \
  --set-file jupyterhub.hub.extraConfig.primehub=https://raw.githubusercontent.com/InfuseAI/primehub/master/helm/primehub/jupyterhub_primehub.py
```

### uninstall

```
helm delete primehub --purge

# remove a unmanaged secret
kubectl -n primehub delete secret primehub-secret
```

**Comment by popcornylu** [comment]
Idempotent is implemented in #19
